test/Microsoft.Framework.PackageManager.FunctionalTests/DnuListTestContext.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Framework.FunctionalTestUtils;
+
+namespace Microsoft.Framework.PackageManager.FunctionalTests
+{
+    public class DnuListTestContext : IDisposable
+    {
+        private readonly IDictionary<Tuple<string, string, string>, DisposableDir> _runtimeHomeDirs =
+            new Dictionary<Tuple<string, string, string>, DisposableDir>();
+
+        private readonly DisposableDir _contextDir;
+
+        public DnuListTestContext()
+        {
+            _contextDir = TestUtils.CreateTempDir();
+            PackageSource = Path.Combine(_contextDir.DirPath, "packages");
+            Directory.CreateDirectory(PackageSource);
+
+            CreateNewPackage("alpha", "0.1.0");
+        }
+
+        public string GetRuntimeHome(string flavor, string os, string architecture)
+        {
+            DisposableDir result;
+            if (!_runtimeHomeDirs.TryGetValue(Tuple.Create(flavor, os, architecture), out result))
+            {
+                result = TestUtils.GetRuntimeHomeDir(flavor, os, architecture);
+                _runtimeHomeDirs.Add(Tuple.Create(flavor, os, architecture), result);
+            }
+
+            return result.DirPath;
+        }
+
+        public string PackageSource { get; }
+
+        private void CreateNewPackage(string name, string version)
```

Comment:
This seems like it should be separate from this class, a static method somewhere so that other tests can use it. Right now this class is specific to DnuList.